tapdb/sqlc/migrations/000033_script_key_type.up.sql

```diff
@@ -0,0 +1,8 @@
+-- The key_type column is used to store the type of key that is stored in the
+-- script_keys table. The type is a Golang numeric type that will have values
+-- such as BIP-0086, script path with custom (externally defined) script, script
+-- path with Taproot Asset Channel related script, etc. The NULL value
+-- will mean the type is not known. Existing script keys will be inspected by
+-- the post-migration step Golang code that determines the type of each script
+-- key.
+ALTER TABLE script_keys ADD COLUMN key_type SMALLINT;
```

Comment:
Non-blocking, but an alternative here would be the enum table pattern. So a new `script_key_type` table w/ set values (burn, transfer, channel, etc) and this new column refs that. This let's us put the mapping of string type to integer within the db itself. 

Tradeoff is potentially a join to match thing sup if just the primary key is stored here. Adding new types also then is a migration, albeit just one that inserts a single row into this new table. 

Just using an integer type here and putting the mapping in the Go layer as is don here certainly simpler.

Reply:
Right, but what would the benefits of that be? IMO it doesn't make the data more easy to read "by eye", meaning that without the join it would still be a non-self-explanatory number.
And on top of that we'd need to string matching to map the values to Golang constants. Or would the idea be to use string-based constants on the Golang level?